subsys/net/lib/downloader/src/dl_socket.c

```diff
@@ -0,0 +1,408 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#if defined(CONFIG_POSIX_API)
+#include <zephyr/posix/sys/socket.h>
+#else
+#include <zephyr/net/socket.h>
+#endif
+#include <zephyr/net/socket_ncs.h>
+#include <zephyr/net/tls_credentials.h>
+#include <stdio.h>
+#include <stdbool.h>
+
+#include "dl_socket.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(downloader, CONFIG_DOWNLOADER_LOG_LEVEL);
+
+#define SIN6(A) ((struct sockaddr_in6 *)(A))
+#define SIN(A) ((struct sockaddr_in *)(A))
+
+#define HOSTNAME_SIZE CONFIG_DOWNLOADER_MAX_HOSTNAME_SIZE
+
+static const char *str_family(int family)
+{
+	switch (family) {
+	case AF_UNSPEC:
+		return "Unspec";
+	case AF_INET:
+		return "IPv4";
+	case AF_INET6:
+		return "IPv6";
+	default:
+		__ASSERT(false, "Unsupported family");
+		return "Unknown";
+	}
+}
+
+int dl_socket_send_timeout_set(int fd, int timeout_ms)
+{
+	int err;
+
+	if (timeout_ms <= 0) {
+		return 0;
+	}
+
+	struct timeval timeo = {
+		.tv_sec = (timeout_ms / 1000),
+		.tv_usec = (timeout_ms % 1000) * 1000,
+	};
+
+	err = setsockopt(fd, SOL_SOCKET, SO_SNDTIMEO, &timeo, sizeof(timeo));
+	if (err) {
+		LOG_WRN("Failed to set socket timeout, errno %d", errno);
+		return -errno;
+	}
+
+	return 0;
+}
+
+static int socket_sectag_set(int fd, const int * const sec_tag_list, uint8_t sec_tag_count)
+{
+	int err;
+	int verify;
+
+	enum {
+		NONE = 0,
+		OPTIONAL = 1,
+		REQUIRED = 2,
+	};
+
+	verify = REQUIRED;
+
+	err = setsockopt(fd, SOL_TLS, TLS_PEER_VERIFY, &verify, sizeof(verify));
+	if (err) {
+		LOG_ERR("Failed to setup peer verification, errno %d", errno);
+		return -errno;
+	}
+
+	LOG_INF("Setting up TLS credentials, sec tag count %u", sec_tag_count);
+	err = setsockopt(fd, SOL_TLS, TLS_SEC_TAG_LIST, sec_tag_list,
+			 sizeof(sec_tag_t) * sec_tag_count);
+	if (err) {
+		LOG_ERR("Failed to setup socket security tag list, errno %d", errno);
+		return -errno;
+	}
+
+	return 0;
+}
+
+static int socket_tls_hostname_set(int fd, const char * const host)
```

Comment:
```suggestion
static int socket_tls_hostname_set(int fd, const char * const hostname)
```

For consistency since there are other functions that have a hostname parameter named `hostname`.